airbyte-cdk/python/airbyte_cdk/sources/abstract_source.py

```diff
@@ -109,22 +109,29 @@ def read(
         with create_timer(self.name) as timer:
             for configured_stream in catalog.streams:
                 stream_instance = stream_instances.get(configured_stream.stream.name)
-                if not stream_instance:
-                    if not self.raise_exception_on_missing_stream:
-                        continue
+                is_stream_exist = True
+                try:
+                    if not stream_instance:
+                        if not self.raise_exception_on_missing_stream:
+                            yield stream_status_as_airbyte_message(configured_stream.stream, AirbyteStreamStatus.INCOMPLETE)
```

Comment:
shouldn't `is_stream_exist` be set to False when executing this block?

It might be easier to directly set `is_stream_exist = bool(stream_instance)`

Reply:
agree
updated